sangyeob/app.js

```diff
@@ -51,11 +50,179 @@ app.post('/users', async (req, res) => {
             [name, email, password, profileImage]
         );
         return res.status(201).json({ message: 'user successfully created' });
-    } catch {
-        return res.status(409).json({ message: 'user email is already taken' });
+    } catch (err) {
+        if (err.errno === 1048) {
+            return res.status(409).json({ error: 'invalid input' });
+        } else if (err.sqlMessage.includes('Duplicate entry')) {
+            return res
+                .status(409)
+                .json({ error: 'user email is already taken' });
+        } else {
+            return res.status(520).json({ error: err.sqlMessage });
+        }
     }
 });
 
+app.post('/posts', async (req, res) => {
+    const { title, content, contentImage, userId } = req.body;
+
+    try {
+        await database.query(
+            `INSERT INTO posts(
+                title,
+                content,
+                content_image,
+                user_id
+            ) VALUES (?,?,?,?);
+            `,
+            [title, content, contentImage, userId]
+        );
+        return res.status(201).json({ message: 'post successfully created' });
+    } catch (err) {
+        if (err.sqlMessage.includes('foreign key constraint fails')) {
+            return res.status(409).json({ error: 'no such user' });
+        } else {
+            return res.status(520).json({ error: err.sqlMessage });
+        }
+    }
+});
+
+app.get('/posts', async (req, res) => {
+    await database.query(
+        `SELECT
+            users.id as userId,
+            users.profile_image as userProfileImage,
+            posts.id as postingId,
+            posts.content_image as postingImageUrl,
+            posts.content as postingContent
+        FROM users
+        INNER JOIN posts ON posts.user_id = users.id
+        `,
+        (err, posts) => {
+            return res.status(200).json(posts);
+        }
+    );
+});
+
+app.get('/users/:userId/posts', async (req, res) => {
+    const { userId } = req.params;
+
+    const obj = {};
+    obj.userId = userId;
+    try {
+        let userProfileImage = await database.query(
+            `SELECT
+                profile_image
+            FROM users 
+            WHERE id = ${userId};
+            `
+        );
```

Comment:
```suggestion
        let userProfileImage = await database.query(
            `SELECT
                profile_image
            FROM users 
            WHERE id = ?;
            `, [ userId]
        );
```

위와 같이 typeorm의 initialize를 통해 만들어진 객체의 .query method에는 SQL 문이 들어갈 수 있습니다.

그 안에 작성되는 SQL 문에는 경찬님이 작성해주신 것과 같이 자리 표시자(${ })를 통해 값을 넣을 수도, 혹은 제가 suggestion으로 작성드린 SQL 안에 ? 를 지정하여 동적으로 쿼리문이 수행될 수 있도록 하는 Placeholder를 사용하는 방법으로 값을 넣을 수도 있습니다.

두가지 방법 모두 다 SQL 안에 입력값을 넣어줄 수 있습니다.

다만 제가 작성한 것과 같이 ?를 활용하여 Placeholder를 사용하는 것을 지향해야 하는 이유는
보안적인 측면과 관련있습니다.

해당 내용은 바로 위 코멘트에서 말씀드린 SQL Injection 같은 case 때문이에요.

따라서 가능하면 ?를 사용하고 placeholder를 통해 값을 넣어주는 방식으로 구현해주세요!